library/swift/src/stats/Counter.swift

```diff
@@ -1,24 +1,15 @@
-@_implementationOnly import EnvoyEngine
 import Foundation
 
 /// A time series counter.
-@objcMembers
-public class Counter: NSObject {
-  private let series: String
-  private weak var engine: EnvoyEngine?
-
-  init(elements: [Element], engine: EnvoyEngine) {
-    self.series = elements.map { $0.value }.joined(separator: ".")
-    self.engine = engine
-    super.init()
-  }
-
+@objc
+public protocol Counter: AnyObject {
   /// Increment the counter by the given count.
-  public func increment(count: Int = 1) {
-    guard let engine = self.engine else {
-      return
-    }
+  func increment(count: Int)
```

Comment:
no default argument value can be specified in Swift's protocol - that's why I need to have a method defined in an extension.